spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -248,37 +229,76 @@ static void validateFeatureGeneratorConfig(Schema inputSchema, String outputColu
   }
 
   /**
-   * Get the valid of input field which is to be used for text based feature generation.
+   * Validate the input field to be used for text based feature generation.
    *
-   * @param input      input Structured Record
+   * @param inputSchema input schema coming in from the previous stage
+   * @param key text field on which to perform text based feature generation
+   */
+  static void validateTextField(Schema inputSchema, String key) {
+    if (inputSchema.getField(key) == null) {
+      throw new IllegalArgumentException(String.format("Input field %s does not exist in the input schema %s.",
+                                                       key, inputSchema.toString()));
+    }
+    Schema schema = inputSchema.getField(key).getSchema();
+    Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+    if (type.equals(Schema.Type.ARRAY)) {
+      Schema componentSchema = schema.getComponentSchema();
+      type = componentSchema.isNullable() ? componentSchema.getNonNullable().getType() :
+        componentSchema.getType();
+    }
+    if (!(type.equals(Schema.Type.STRING) || type.equals(Schema.Type.ARRAY))) {
```

Comment:
this should just be `type != Schema.Type.STRING` (you can compare directly for enums). It is not valid for it to be an array. If it is, it means the field is an array of arrays.